File: src/com/sterndu/util/MathUtil.java
package com.sterndu.util;

import java.math.BigInteger;

public class MathUtil {

	public static int avg(int o, int r) {
		double doi = o;
		doi+=r;
		doi/= 2;
		return (int) Math.round(doi);
	}

	public static BigInteger factorial(long x) {
		BigInteger y = BigInteger.ONE;
		x = Math.abs(x);
		for (; x > 1; x--) {
			y = y.multiply(BigInteger.valueOf(x));
		}
		return y;
	}

}

File: src/com/sterndu/util/MathUtil.kt
@file:JvmName("MathUtil")
package com.sterndu.util

import java.math.BigInteger
import kotlin.math.abs
import kotlin.math.roundToInt

object MathUtil {
	fun avg(o: Int, r: Int): Int {
		return intArrayOf(o, r).average().roundToInt()
	}

	fun factorial(initialX: Long): BigInteger {
		var x = abs(initialX)
		var y = BigInteger.ONE
		while (x > 1) {
			y = y.multiply(BigInteger.valueOf(x))
			x--
		}
		return y
	}
}
